pulser/waveforms.py

```diff
@@ -723,10 +812,26 @@ def change_duration(self, new_duration: int) -> InterpolatedWaveform:
         return InterpolatedWaveform(new_duration, self._values, **self._kwargs)
 
     def _plot(
-        self, ax: Axes, ylabel: str, color: Optional[str] = None
+        self,
+        ax: Axes,
+        ylabel: Optional[str] = None,
+        color: Optional[str] = None,
+        channel: Optional[Channel] = None,
+        label: str = "",
+        start_t: int = 0,
     ) -> None:
-        super()._plot(ax, ylabel, color=color)
-        ax.scatter(self._data_pts[:, 0], self._data_pts[:, 1], c=color)
+        super()._plot(
+            ax,
+            ylabel,
+            color=color,
+            channel=channel,
+            label=label,
+            start_t=start_t,
+        )
+        if not channel:
```

Comment:
`channel` being given is a proxy for ploting the output samples. The points are already being plotted with the input (when `channel` is not given), so this would just repeat that with a different color.